qiskit/algorithms/minimum_eigensolvers/vqe.py

```diff
@@ -0,0 +1,380 @@
+            try:
+                logger.info(
+                    "Trying to resize ansatz to match operator on %s qubits.", operator.num_qubits
+                )
+                ansatz.num_qubits = operator.num_qubits
+            except AttributeError as error:
+                raise AlgorithmError(
+                    "The number of qubits of the ansatz does not match the "
+                    "operator, and the ansatz does not allow setting the "
+                    "number of qubits using `num_qubits`."
+                ) from error
+
+        if ansatz.num_parameters == 0:
+            raise AlgorithmError("The ansatz must be parameterized, but has no free parameters.")
+
+        return ansatz
+
+    def _eval_aux_ops(
+        self,
+        ansatz: QuantumCircuit,
+        aux_operators: ListOrDict[BaseOperator | PauliSumOp],
+    ) -> ListOrDict[tuple(complex, complex)]:
+        """Compute auxiliary operator eigenvalues."""
+
+        if isinstance(aux_operators, dict):
+            aux_ops = list(aux_operators.values())
+        else:
+            aux_ops = aux_operators
+
+        num_aux_ops = len(aux_ops)
+        aux_job = self.estimator.run([ansatz] * num_aux_ops, aux_ops)
+        aux_values = aux_job.result().values
+        aux_values = list(zip(aux_values, [0] * len(aux_values)))
+
+        if isinstance(aux_operators, dict):
+            aux_values = dict(zip(aux_operators.keys(), aux_values))
+
+        return aux_values
+
+
+def _validate_initial_point(point, ansatz):
+    expected_size = ansatz.num_parameters
+
+    if point is None:
+        # get bounds if ansatz has them set, otherwise use [-2pi, 2pi] for each parameter
+        bounds = getattr(ansatz, "parameter_bounds", None)
+        if bounds is None:
+            bounds = [(-2 * np.pi, 2 * np.pi)] * expected_size
+
+        # replace all Nones by [-2pi, 2pi]
+        lower_bounds = []
+        upper_bounds = []
+        for lower, upper in bounds:
+            lower_bounds.append(lower if lower is not None else -2 * np.pi)
+            upper_bounds.append(upper if upper is not None else 2 * np.pi)
+
+        # sample from within bounds
+        point = algorithm_globals.random.uniform(lower_bounds, upper_bounds)
+
+    elif len(point) != expected_size:
+        raise ValueError(
+            f"The dimension of the initial point ({len(point)}) does not match the "
+            f"number of parameters in the circuit ({expected_size})."
+        )
+
+    return point
+
+
+class VQEResult(VariationalResult, MinimumEigensolverResult):
+    """Variational quantum eigensolver result."""
+
+    def __init__(self) -> None:
+        super().__init__()
+        self._cost_function_evals = None
+
+    @property
+    def cost_function_evals(self) -> int | None:
+        """Returns number of cost optimizer evaluations."""
+        return self._cost_function_evals
+
+    @cost_function_evals.setter
+    def cost_function_evals(self, value: int) -> None:
```

Comment:
I don't think the type is right here. It returns a list of int and the setter is a list (numpy arrays). I guess it was not noticed on the original but you can see above its a list, the optimizer evals for each eigenvalue found.